src/vaadin-chart-series.html

```diff
@@ -266,117 +270,107 @@
+            series.yAxis.update({[key]: value});
           }
         }
 
-        __valueMaxObserver() {
-          if (!this.__hasSeriesConfig()) {
+        __valueMinObserver(valueMin, series) {
+          if (valueMin === undefined || series === undefined) {
             return;
           }
 
-          if (!isFinite(this.valueMax)) {
-            this.__showWarn('value-max', 'Numbers or null');
+          this.__updateAxis(series, valueMin, 'min');
+        }
+
+        __valueMaxObserver(valueMax, series) {
+          if (valueMax === undefined || series === undefined) {
             return;
           }
 
-          if (this._series.yAxis) {
-            this._series.yAxis.update({
-              max: this.valueMax
-            });
-          }
+          this.__updateAxis(series, valueMax, 'max');
         }
 
-        __typeObserver() {
-          if (this.__hasSeriesConfig()) {
-            this._series.update({
-              type: this.type
-            });
+        __typeObserver(type, series) {
+          if (type && series) {
+            series.update({type});
           }
         }
 
-        __titleObserver() {
-          if (this.__hasSeriesConfig()) {
-            this._series.update({
-              name: this.title
-            });
+        __titleObserver(title, series) {
+          if (title === undefined || series === undefined) {
+            return;
           }
+          series.update({name: title});
         }
 
-        __stackObserver() {
-          if (!this.__hasSeriesConfig()) {
+        __stackObserver(stack, series) {
+          if (stack === undefined || series === undefined) {
             return;
           }
-
-          this._series.update({
-            stack: this.stack
-          });
+          series.update({stack});
         }
 
-        __neckPositionOberserver() {
-          if (!this.__hasSeriesConfig()) {
+        __neckPositionObserver(neckPosition, series) {
+          if (neckPosition === undefined || series === undefined) {
             return;
           }
-          this._series.update({
-            neckHeight: this.neckPosition
-          });
+
+          series.update({neckHeight: neckPosition});
         }
 
-        __neckWidthOberserver() {
-          if (!this.__hasSeriesConfig()) {
+        __neckWidthObserver(neckWidth, series) {
+          if (neckWidth === undefined || series === undefined) {
             return;
           }
-          this._series.update({
-            neckWidth: this.neckWidth
-          });
+
+          series.update({neckWidth});
         }
 
-        __unitObserver() {
-          if (this.__hasSeriesConfig()) {
+        __unitObserver(unit, valueMin, valueMax, series) {
+          if (series && unit !== this.__oldUnit) {
+            this.__oldUnit = unit;
+
             const parent = this.parentNode instanceof Vaadin.ChartElement && this.parentNode;
             if (parent && parent instanceof Vaadin.ChartElement) {
```

Comment:
I know this is not part of the refactor, but this second check on the if doesn't make sense to me or I am missing something.

Reply:
Thanks, noted. I'd like to merge this PR to unlock #475 so I will fix this separately.